### ISSUE TITLE
[l10n] add bs language

### ISSUE THREAD

**Issue by AndyAnds265** [brave/brave-browser#37196]
Hello,

We would like to begin supporting the Bosnian language which is indicated by the locale code `bs`. 

We will not be paying for localization services in Bosnian; however we'll be leveraging a combination of resources from Chromium and previously existing translations. 

For context, Bosnian, Croatian, Montenegrin and Serbian languages are all very similar languages. In fact, they are fundamentally the same language with some political differences due to their history. 

Currently we have a substantial user base in Bosnian but as we do not support the locale `bs`, all of those users are viewing the app and content in English. 

Although it isn't a perfect solution, we would like to copy over the Croatian localizations since it is the closest to Bosnian and the least political. 

Therefore, moving forward, can we please use a one-to-many mapping for the `hr` language:

``hr`: bs, hr`

**Comment by AndyAnds265** [comment]
The goal here is to allow us to cover more with less

**Comment by AndyAnds265** [comment]
@mkarolin is this something you can handle?

**Comment by AndyAnds265** [comment]
@deeppandya or maybe Deep?

**Comment by mkarolin** [comment]
I looked into this a bit and it seems like, unfortunately, this is not going to be useful. Currently, Chrome (and consequently Brave) doesn't allow to set Bosnian as the UI language on any of the Desktop platforms or on Android (@deeppandya, please let me know if I am wrong). You can add Bosnian to the languages, but on Desktop the option to set it as the UI language is greyed out. On Android, I could set Bosnian as my UI language but it still showed in English, perhaps the entire phone's preferred language needs to be changed, but I have not attempted that. As to why Chrome doesn't support Bosnian as a UI language is not clear to me, perhaps it has something to do with the native platform support for the language (https://source.chromium.org/chromium/chromium/src/+/main:build/config/locales.gni;l=132)

**Comment by deeppandya** [comment]
yeah we can select `Bosniak` as one of the language for android. and it seems chromium has translations for `Bosniak` so it makes sense to include it for brave. @AndyAnds265 do we know how many users we have in the region ?

**Comment by AndyAnds265** [comment]
At the moment, we have 7.7k users in Bosnian

![Screenshot 2024-04-18 at 4 14 07 PM](https://github.com/brave/brave-browser/assets/19779922/0a39000f-1ef3-41d9-afef-38eb3b86a35c)

**Comment by deeppandya** [comment]
@Andy we seem to be missing the support from chromium even though they have the translation file. i also confirmed that with chrome app. changing the language in OS, doesn't do anything for Bosniak.

**Comment by Susexe** [comment]
Please also take a look at [this issue](https://github.com/brave/brave-browser/issues/35417).

**Comment by AndyAnds265** [comment]
@deeppandya I confirmed by looking in the Chromium files that `bs` is present. You can see the language here: https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/android/strings/translations/android_chrome_strings_bs.xtb

Since the language is there, is there a way we can contact someone at the Chromium dev side or something? It could be possible that there's a bug in a few languages

**Comment by AndyAnds265** [comment]
Just wanted to bump this - any updates?

**Comment by Susexe** [comment]
@AndyAnds265, do we have any updates on this? As previously mentioned, the option to enable the language is grayed out in Chrome. However, I believe the same issue existed for Sinhala, and it was successfully resolved in Brave.

**Comment by bsclifton** [comment]
This appears to be tracked upstream with https://issues.chromium.org/issues/40679433

However, it was previously closed as `wontfix`. I don't think this is a high priority for the Chromium project

There are also these two open issues:
- [[more locales] Intl.DateTimeFormat - Incorrect date formatting for "bs" Bosnian locale](https://issues.chromium.org/issues/40172622)
- [[more locales] Intl.NumberFormat - Incorrect currency formatting for "bs" Bosnian locale](https://issues.chromium.org/issues/41475399)

**Comment by bsclifton** [comment]
I think this request by @AndyAnds265 is basically asking if we can copy/paste the locale resources for the `hr` language and rename the copy as `bs`. And wire up the UI so it uses those with a new option `Bosnian`. 

cc: @samartnik @AlexeyBarabash - not sure how difficult that would be to maintain between Chromium upgrades

**Comment by Susexe** [comment]
@bsclifton Please note that localization files for Bosnian already exist; they're just not being used, so there's no need to copy Croatian translations, just enable the locale.

**Comment by mkarolin** [comment]
**TEST PLAN**

**Desktop**
1. Start Brave
2. Navigate to brave://settings/languages
3. In the `Preferred languages` section click on `Add languages` button
4. In the search bar start typing "Bosnian"
5. Check the box and click Add
6. Bosnian should appear as an entry in the preferred languages section
7. Click on the tree dots menu for Bosnian and check Display Brave in this language box
8. Click on the restart button
9. Browser restarts

Expected result:
Browser UI is shown in Bosnian language.

**Android**
1. Start Brave
2. Three dots menu -> Settings -> Languages
3. Click on selected language (likely Current device language)
4. In the search bar type Bosnian and when found click on it

Expected result: downloading message will show up on the languages page
Expected result: once the language pack downloads the browser UI switches to Bosnian language.

**Comment by kjozwiak** [comment]
The above requires `1.92.115` or higher for `1.92.x` verification(s) 👍

**Comment by Uni-verse** [comment]
Verified on `Samsung Galaxy Tab S7` using the following version: 

```
Brave	1.92.134 Chromium: 150.0.7871.63 (Official Build) (64-bit) 
Revision	fcbe40ead62db38cdfb5e995e215a1b9fa4a28ae
OS	Android 13; Build/TP1A.220624.014; 33; REL
```
- Sanity tested support for Bosnian (bs) language. 
   - [x] onboarding
   - [x] menus, settings, dropdowns 
   - [x] Shields, Rewards, Sync
   - [x] paywalls
   - [x] Leo, VPN
   - [x] tabs, context menus, bookmarks

example | example | example | example
--|--|--|--
<img width="873" height="1359" alt="Image" src="https://github.com/user-attachments/assets/7be09a4e-5822-44ac-8495-d9e607330457" /> | <img width="873" height="1359" alt="Image" src="https://github.com/user-attachments/assets/72724d70-8fa6-4d0b-a387-5bb554b60549" /> | <img width="873" height="1359" alt="Image" src="https://github.com/user-attachments/assets/e4f263c0-8683-42e5-906c-327529cf1412" /> | <img width="873" height="1359" alt="Image" src="https://github.com/user-attachments/assets/bf73b2dd-a144-4bcd-bed6-800bf8f27815" />
<img width="873" height="1359" alt="Image" src="https://github.com/user-attachments/assets/9f48bfb0-917d-4959-997a-f03bc2b7ff75" /> | <img width="873" height="1359" alt="Image" src="https://github.com/user-attachments/assets/36e35468-5e1a-4dff-ac19-6b0d17fc8cd5" /> | <img width="873" height="1359" alt="Image" src="https://github.com/user-attachments/assets/b61b7d9b-8c6d-40ea-aac8-7c1b09d96ed8" /> | <img width="873" height="1359" alt="Image" src="https://github.com/user-attachments/assets/19bf6cd6-9c87-425a-8e7c-f8b4a81f7a34" />
<img width="873" height="1359" alt="Image" src="https://github.com/user-attachments/assets/374dbafe-31cc-416b-a4f2-6d382cdb55ff" /> | <img width="873" height="1359" alt="Image" src="https://github.com/user-attachments/assets/4d532985-af8e-4a8f-a8b7-f311deac02cc" /> | <img width="873" height="1359" alt="Image" src="https://github.com/user-attachments/assets/840c4b47-3bd3-443a-9c6c-c3c70730fdbf" /> | <img width="873" height="1359" alt="Image" src="https://github.com/user-attachments/assets/06558933-74b7-4842-b5e4-15778f8b26b6" />
<img width="873" height="1359" alt="Image" src="https://github.com/user-attachments/assets/755c9fda-8897-40f4-a9ce-6d5c4934bc9c" /> | <img width="873" height="1359" alt="Image" src="https://github.com/user-attachments/assets/550b9c2c-dba4-4713-894c-f16a0863f772" /> | <img width="873" height="1359" alt="Image" src="https://github.com/user-attachments/assets/4a6134e1-01e5-48b7-9eb8-276bf3adb024" /> | <img width="873" height="1359" alt="Image" src="https://github.com/user-attachments/assets/7bd19e74-3784-4bea-8a2b-6109552bfc8f" />
